### ISSUE TITLE
Make it clear that summary is mandatory for Wikibase upload

### ISSUE THREAD

**Issue by cooperzoe** [OpenRefine/OpenRefine#6306]
When I was close to finishing my Wikibase edit and needed to add a summary in the upload menu, it wasn't clear to me that the **summary** was mandatory for the upload to work.

Let's add a visual cue - for example, an asterisk - to indicate the field is required.
<img width="1455" alt="Upload edits" src="https://github.com/OpenRefine/OpenRefine/assets/52141007/a4f43452-f65b-4e3d-9f9f-5b3b63c7d6f6">

**Comment by thadguidry** [comment]
I think asterisk is not enough and can be confusing.  We can have a div that includes the asterisk, and div/span next to it with some translatable text element.  We could then reuse this "required" div in many input fields throughout OpenRefine when the field is required.

![image](https://github.com/OpenRefine/OpenRefine/assets/986438/447c02b3-39d0-4dff-ab90-a36a07415235)

**Comment by trnstlntk** [comment]
? It's not mandatory (at least not on Wikibases) so the upload should not fail on the OpenRefine side if the user does not fill in anything.

**Comment by wetneb** [comment]
@thadguidry yes it would be good to have a unified way to mark such fields as required (I also can't remember on top of my mind any field in the UI where we currently signal the fact that it is required visually). It would be useful for instance for the column name field in "add column based on this column".

I can share why I made this field mandatory when first developing this feature (happy to change it). When uploading edits to a Wikibase, OpenRefine tries to minimize the number of edits it makes on each entity. For instance, if you are adding two statements on the same Wikidata item, they will generally be added in a single edit. That saves time and computing resources on the Wikibase instance. As a consequence, the edit summary that is automatically generated by Wikibase for this edit will be something like "Updated Item: ". When reviewing the history of a Wikidata item, those sorts of summaries are very uninformative, so I thought it's good to encourage users to supply a more descriptive summary of their own.
I think it makes sense that the Wikibase UI does not ask for summaries on its side, since it would be very cumbersome to have to supply that for each manual edit, for which the automated summaries will generally work well enough.

**Comment by t8210103** [comment]
Hello, i would like to be assigned this issue. 
Also, according the implementation, should i add a "* required" as suggested, should i add a pre-made standardized message that can be edited (by clicking on the input box as it is right now) and helps the user from having to fill the empty input or should i focus on creating automated summaries (which can also be edited), that represent the edits the user is about to upload?
If i should implement the last proposal, i would like some guidance on how to collect necessary information about the edits, so i can create custom messages.

**Comment by wetneb** [comment]
I would go for marking the field as required. I am not sure if we use the `*` sign anywhere else in the UI, but perhaps it's widespread enough to be understood anyway.

If I remember correctly, when trying to validate the form without a summary, we currently only focus the field but not make it clearer that it's required. In some forms there is an additional explanation that shows up in such a case, such as a "This field is required." in red. Maybe we could also consider that.

**Comment by wetneb** [comment]
I just remembered that HTML actually has a `required` attribute for form inputs which would very well do the job (for implementing the "This field is required" hint):
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required
